core/mobile-manager/src/backend/MobileHost.ts

```diff
@@ -134,8 +99,18 @@ export class MobileHost {
     return new Promise<void>((resolve, reject) => {
 
       let progressCb: MobileProgressCallback | undefined;
+      let lastReportedOn = Date.now();
+      const minTimeBeforeReportingProgress = 1000;
       if (progress) {
         progressCb = (_bytesWritten: number, totalBytesWritten: number, totalBytesExpectedToWrite: number) => {
+          const currentTime = Date.now();
```

Comment:
@khanaffan I think this logic should be moved to `NativeHost.downloadBriefcase`

Reply:
but i don't understand, when is this used? For briefcases? How?

Reply:
This is called by MobileFileHandler. Which is set as file handler in iModelClient which is what is used to download changeset, checkpoints and briefcases.

I can move this logic down a step in native iOS level as it is the one firing up event more quickly then it could be transported to frontend.

Just one question is it possible for IPC notification. Progress kind of events are such that only latest event matters. If previous events are not process by frontend end it is fine to override it with latest. I did implement something similar when event was polled and progress was fired with flag to override previous event so poll will only find one event in queue at anytime for a given type of event.

Reply:
Events are no longer queued for later dispatch, they are dispatched immediately. The frontend no longer polls for events. EventSink and EventSource no longer exist.

Reply:
So i guess this PR can go through as is, i will move this logic in separate PR down to IOS level to reduce calling into node in the first place. Just limit 1 progress event/ sec.

Reply:
@khanaffan i added logic into the NativeHost.downloadBriefcase to only send progress events every 500 milliseconds. That takes care of the download briefcase use, is `MobileFileHandler` used for other types of downloads?

Reply:
Yes it is. It also used for changeset download. But in that case we do not report progress to frontend. We have backlog item to report all progress for each of following step so app can actually show why its taking time to open a  iModel.

1. download briefcase
2. download changeset
3. apply changesets.